compiler/rustc_lint/src/context.rs

```diff
@@ -530,9 +530,7 @@ pub trait LintContext {
         lint: &'static Lint,
         span: Option<impl Into<MultiSpan>>,
         msg: impl Into<DiagnosticMessage>,
-        decorate: impl for<'a, 'b> FnOnce(
-            &'b mut DiagnosticBuilder<'a, ()>,
-        ) -> &'b mut DiagnosticBuilder<'a, ()>,
+        decorate: impl for<'a, 'b> FnOnce(&'b mut DiagnosticBuilder<'a, ()>),
```

Comment:
Oh, well, I guess I missed that. I'm not necessarily convinced by either of the arguments, but yeah, I'll leave it up to waffle to weigh in.

Reply:
> Though I will say that I wonder if the comment is correct, because as written the change seems like an improvement, shortening the code a little. And any function of the form `f(..., &mut T) -> &mut T` is really weird.

The shortening comment refers to is the `|lint| { lint.span_note(""); }` (which auto-formats to 3 lines) vs `|lint| lint.span_note("")` (auto-formats to one). I think I overestimated how common this is.

I don't personally find `&mut T -> &mut T` a weird interface though, it's just the usual builder function (just like `span_note`).

Reply:
`&mut self -> &mut Self` is one of the typical builder patterns, right? So this is similar but not quite the same. Not as weird as I first thought.

Reply:
Yes